streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,287 @@
+                100, // a fixed large enough value will suffice
+                streamsConfig.getLong(streamsConfig.RECONNECT_BACKOFF_MS_CONFIG),
+                streamsConfig.getInt(streamsConfig.SEND_BUFFER_CONFIG),
+                streamsConfig.getInt(streamsConfig.RECEIVE_BUFFER_CONFIG),
+                streamsConfig.getInt(streamsConfig.REQUEST_TIMEOUT_MS_CONFIG), time);
+
+        brokerNode = kafkaClient.leastLoadedNode(new SystemTime().milliseconds());
+    }
+
+    public KafkaClient getKafkaClient() {
+        return kafkaClient;
+    }
+
+    public void shutdown() {
+        log.info("Closing the StreamsKafkaClient.");
+        // this will keep track of the first encountered exception
+        AtomicReference<Throwable> firstException = new AtomicReference<Throwable>();
+        ClientUtils.closeQuietly(metrics, "producer metrics", firstException);
+
+        log.debug("The StreamsKafkaClient has closed.");
+        if (firstException.get() != null)
+            throw new KafkaException("Failed to close kafka producer", firstException.get());
+    }
+
+    /**
+     * Cretes a new topic with the given number of partitions and replication factor.
+     *
+     * @param internalTopicConfig
+     * @param numPartitions
+     * @param replicationFactor
+     */
+    public void createTopic(InternalTopicConfig internalTopicConfig, int numPartitions, int replicationFactor, long windowChangeLogAdditionalRetention) {
+        CreateTopicsRequest.TopicDetails topicDetails;
+        Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
+        Map<String, String> topicConfig = new HashMap<>();
+        for (String key : topicProperties.stringPropertyNames()) {
+            topicConfig.put(key, topicProperties.getProperty(key));
+        }
+        topicDetails = new CreateTopicsRequest.TopicDetails(numPartitions, (short) replicationFactor, topicConfig);
+        Map<String, CreateTopicsRequest.TopicDetails> topics = new HashMap<>();
+        topics.put(internalTopicConfig.name(), topicDetails);
+
+        CreateTopicsRequest createTopicsRequest = new CreateTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        RequestCompletionHandler callback = new RequestCompletionHandler() {
+            public void onComplete(ClientResponse response) {
+                // Do nothing!
+            }
+        };
+        ClientResponse clientResponse = sendRequest(createTopicsRequest.toStruct(), ApiKeys.CREATE_TOPICS, callback);
+        CreateTopicsResponse createTopicsResponse = new CreateTopicsResponse(clientResponse.responseBody());
+        if (createTopicsResponse.errors().get(internalTopicConfig.name()).code() > 0) {
+            throw new StreamsException("Could not create topic: " + internalTopicConfig.name());
+        }
+    }
+
+    /**
+     * Delete a topic.
+     *
+     * @param topic
+     */
+    public void deleteTopic(String topic) {
```

Comment:
It's not being used but we can have it in the StreamsKafkaClient for future use.